scripts/globals/limbus.lua

```diff
@@ -530,141 +531,177 @@ function xi.limbus.openDoor(battlefield, doorID)
     door:setAnimation(xi.animation.OPEN_DOOR)
 end
 
-function xi.limbus.extendTimeLimit(ID, battlefield, amount)
-    local timeLimit = battlefield:getTimeLimit()
-    battlefield:setTimeLimit(timeLimit + amount * 60)
-    local remaining = battlefield:getRemainingTime() / 60
-
-    for _, player in pairs(battlefield:getPlayers()) do
-        player:messageSpecial(ID.text.TIME_EXTENDED, amount)
-        player:messageSpecial(ID.text.TIME_LEFT, remaining)
-    end
-end
-
-LimbusArea = {}
-LimbusArea.__index = LimbusArea
-LimbusArea.__eq = function(m1, m2)
+Limbus = setmetatable({ }, { __index = Battlefield })
+Limbus.__index = Limbus
+Limbus.__eq = function(m1, m2)
     return m1.name == m2.name
 end
 
-LimbusArea.name = ""
-LimbusArea.serverVar = ""
-LimbusArea.requiredCard = 0
-LimbusArea.groups = {}
-LimbusArea.paths = {}
-LimbusArea.loot = {}
+Limbus.name = ""
+Limbus.serverVar = ""
+Limbus.requiredCard = 0
 
-function LimbusArea:new(name, requiredCard)
-    local obj = {}
+function Limbus:new(zoneId, battlefieldId, menuBit, entryNpc, name)
+    local obj = Battlefield:new(zoneId, battlefieldId, menuBit)
     setmetatable(obj, self)
     obj.name = name
+    obj.ID = zones[obj.zoneId][obj.name]
     obj.serverVar = "[" .. name .. "]Time"
-    obj.requiredCard = requiredCard
+    table.insert(obj.requiredKeyItems, xi.ki.COSMO_CLEANSE)
+    obj:setEntryNpc(entryNpc)
+    obj.createWornItem = false
     return obj
 end
 
-function LimbusArea:createBattlefield()
-    local battlefield_object = {}
-
-    battlefield_object.onBattlefieldInitialise = function(battlefield)
-        battlefield:setLocalVar("loot", 1)
-        SetServerVariable(self.serverVar, battlefield:getTimeLimit() / 60)
+function Limbus:onEntryEventUpdate(player, csid, option, extras)
+    print("Limbus:onEntryEventUpdate")
+    if Battlefield.onEntryEventUpdate(self, player, csid, option, extras) then
+        local alliance = player:getAlliance()
+
+        for _, member in pairs(alliance) do
+            if
+                member:getZoneID() == player:getZoneID() and
+                not member:hasStatusEffect(xi.effect.BATTLEFIELD) and
+                not member:getBattlefield()
+            then
+                member:messageSpecial(self.ID.text.HUM)
+            end
+        end
+    end
+end
 
-        battlefield:addGroups(self.groups)
+function Limbus:onEntryEventFinish(player, csid, option)
+    if not Battlefield.onEntryEventFinish(self, player, csid, option) then
+        player:setCharVar("ApollyonEntrance", 0)
+    end
+end
 
-        -- Setup Paths
-        for mobId, path in pairs(self.paths) do
-            GetMobByID(mobId):pathThrough(path, xi.path.flag.PATROL)
-        end
+function Limbus:onEntryTrade(player, npc, trade)
+    if self:checkRequirements(player, npc, true) then
+        player:setCharVar("ApollyonEntrance", 1)
```

Comment:
Just a note on this `ApollyonEntrance` var: 1 signifies that they entered one of the eastern zones and is used in the zones onEventFinish to set their pos to the eastern side when they exit since the cutscene always spits them out on the western side. There's likely some option that needs to be set for the cutscene instead but I couldn't figure it out at the time.

Reply:
Thanks! I noticed that after the player zones out they get shifted over. I'll be sure to look into that.

Reply:
Got that fixed! https://github.com/LandSandBoat/server/pull/2824/commits/67bcd5b2544c010e245834b72c375a014f693b6d